src/common/dts.c

```diff
@@ -270,7 +270,9 @@ pool_fini(struct dts_context *tsc)
 			DP_RC(rc));
 
 	} else { /* DAOS mode */
-		daos_pool_disconnect(tsc->tsc_poh, NULL);
+		rc = daos_pool_disconnect(tsc->tsc_poh, NULL);
```

Comment:
This could fail for any number of reasons, for example -DER_NOMEM, is assert the right thing here?

For test code this would be acceptable, however as a general rule assert() is not appropriate for error checking so if this code is in either the client or the server it needs to check and propagate the error.  I'm confused because this file doesn't have "test" in the name anywhere but is also using MPI so it's not clear where this source is used.